app/components/assets/actions-dropdown.tsx

```diff
@@ -95,126 +98,166 @@ const ConditionalActionsDropdown = () => {
           ref={dropdownRef}
         >
           <div className="order fixed bottom-0 left-0 w-screen rounded-b-none rounded-t-[4px] bg-white p-0 text-right md:static md:w-[180px] md:rounded-t-[4px]">
-            <DropdownMenuItem
-              className="border-b px-4 py-1 md:p-0"
-              disabled={assetIsCheckedOut && !assetCanBeReleased}
-            >
-              {assetCanBeReleased ? (
-                <Button
-                  to="overview/release-custody"
-                  role="link"
-                  variant="link"
-                  className={tw(
-                    "justify-start whitespace-nowrap px-4 py-3  text-gray-700 hover:text-gray-700"
-                  )}
-                  width="full"
-                  onClick={() => setOpen(false)}
-                  disabled={assetIsPartOfUnavailableKit}
+            {!isSelfService ? (
```

Comment:
Here instead of doing this I would do the <When> component 2 times using `userHasPermissions`

1.  First time will be for the update action
2. Second time will be for the custody actions


I would rather use `userHasPermissions` as the single source of truth and only in some special scenarios use `isSelfService`